doc/collections_as_container.md

```diff
@@ -167,17 +167,57 @@ In addition to the *LegacyForwardIterator* the C++ standard specifies also the *
 | `std::back_insert_iterator` | ❗ attention | Compatible only with SubsetCollections, otherwise throws `std::invalid_argument` |
 | `std::front_insert_iterator` | ❌ no | `push_front` not defined |
 | `std::insert_iterator` | ❌ no | `insert` not defined |
-| `std::const_iterator` | ❌ no | `iterator` and `const_iterator` not *LegacyInputIterator* or `std::input_iterator` |
-| `std::move_iterator` | ❌ no | `iterator` and `const_iterator` not *LegacyInputIterator* or `std::input_iterator` |
-| `std::counted_iterator` |  ❌ no | `iterator` and `const_iterator` not `std::input_or_output_iterator` |
-
+| `std::const_iterator` (C++23) | ❌ no | C++23 not supported yet |
```

Comment:
Is this only an issue of enabling c++23 in our cmake config, here?

https://github.com/AIDASoft/podio/blob/f6662ad9bf5c202f487b13d70914b5f0b6f29e2a/CMakeLists.txt#L39-L41

Reply:
I was a bit lazy and wanted to skip that part :sweat_smile:
Added checks. It seems that `std::const_iterator` should be fine except that `->` can't be used as `->` in our iterators can't be `const`

Reply:
Ah sorry, I didn't want to imply more work. I was mainly wondering because of the comment and if that would have been the only reason, it would have been an easy fix.

Reply:
no problem, thanks to this I learned some more new C++23 stuff :slightly_smiling_face: